cobc/typeck.c

```diff
@@ -6280,14 +6295,14 @@ decimal_expand (cb_tree d, cb_tree x)
 	/* skip if the actual statement can't be generated any more
 	   to prevent multiple errors here */
 	if (error_statement == current_statement) {
-		return;
+		return d;
 	}
 	switch (CB_TREE_TAG (x)) {
 	case CB_TAG_CONST:
 		/* LCOV_EXCL_START */
 		if (x != cb_zero) {
 			cobc_err_msg (_("unexpected constant expansion"));
-			COBC_ABORT ();
```

Comment:
That, as with (nearly?) any call of `cobc_err_msg()` should lead to an abort.
Did you got into this (the testcase don't contain an example)? If yes: what was the content of `x`?
If not: please revert that.

Reply:
Indeed, no, I didn't get into this error, I just thought it was aborting because there was no way to return an error, but I can put back the previous behavior.